go/vt/vttablet/tabletmanager/healthcheck.go

```diff
@@ -242,6 +242,7 @@ func (agent *ActionAgent) runHealthCheckLocked() {
 			if healthErr == nil {
 				// we were unhealthy, are now healthy,
 				// make sure we have the right mysql port.
+				agent.gotMysqlPort = false
 				if updatedTablet := agent.checkTabletMysqlPort(agent.batchCtx, tablet); updatedTablet != nil {
```

Comment:
Yes, this case is covered. In fact, that's how I found this code path: I modified the test after I changed the code below to set the flag, and it didn't work.

However, the case below is not covered, let me see if I can cover it somehow.